cmd/juju/cloud/addcredential_test.go

```diff
@@ -207,6 +209,36 @@ func (s *addCredentialSuite) TestAddCredentialMultipleAuthType(c *gc.C) {
 	s.assertAddUserpassCredential(c, "fred\nuserpass\nuser\npassword\n", nil)
 }
 
+func (s *addCredentialSuite) TestAddCredentialInteractive(c *gc.C) {
+	s.authTypes = []jujucloud.AuthType{"interactive"}
+	s.schema = map[jujucloud.AuthType]jujucloud.CredentialSchema{
+		"interactive": {{"username", jujucloud.CredentialAttr{}}},
+	}
+
+	stdin := strings.NewReader("fred\nuser\n")
+	ctx, err := s.run(c, stdin, "somecloud")
+	c.Assert(err, jc.ErrorIsNil)
+
+	c.Assert(testing.Stderr(ctx), gc.Equals, `
+Enter credential name: Using auth-type "interactive".
+Enter username: generating userpass credential
+`[1:])
+
+	// FinalizeCredential should have generated a userpass credential
+	// based on the input from the interactive credential.
+	c.Assert(s.store.Credentials, jc.DeepEquals, map[string]jujucloud.CloudCredential{
+		"somecloud": {
+			AuthCredentials: map[string]jujucloud.Credential{
+				"fred": jujucloud.NewCredential(jujucloud.UserPassAuthType, map[string]string{
+					"username":             "user",
```

Comment:
I know it's a test and this is test data but username = "user" caught me off-guard \o/
Could this be "username" too for clarity?